Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,588 @@
+          align: 'left',
+        },
+      };
+
+      const MultiStyleText = objectsRenderingService.requireModule(
+        __dirname,
+        'pixi-multistyle-text/dist/pixi-multistyle-text.umd'
+      );
+
+      this._pixiObject = new MultiStyleText('', BBTextStyles);
+
+      this._pixiObject.anchor.x = 0.5;
+      this._pixiObject.anchor.y = 0.5;
+      this._pixiContainer.addChild(this._pixiObject);
+      this.update();
+    }
+    RenderedBBTextInstance.prototype = Object.create(
+      RenderedInstance.prototype
+    );
+
+    /**
+     * Return the path to the thumbnail of the specified object.
+     */
+    RenderedBBTextInstance.getThumbnail = function(
+      project,
+      resourcesLoader,
+      object
+    ) {
+      return 'JsPlatform/Extensions/bbcode24.png';
+    };
+
+    /**
+     * This is called to update the PIXI object on the scene editor
+     */
+    RenderedBBTextInstance.prototype.update = function() {
+      const rawText = this._associatedObject
+        .getProperties(this.project)
+        .get('text')
+        .getValue();
+      if (rawText !== this._pixiObject.text) {
+        this._pixiObject.setText(rawText);
+      }
+
+      const opacity = this._associatedObject
+        .getProperties(this.project)
+        .get('opacity')
+        .getValue();
+      this._pixiObject.alpha = opacity / 255;
+
+      const color = this._associatedObject
+        .getProperties(this.project)
+        .get('color')
+        .getValue();
+      this._pixiObject.textStyles.default.fill = color;
+
+      const fontSize = this._associatedObject
+        .getProperties(this.project)
+        .get('fontSize')
+        .getValue();
+      this._pixiObject.textStyles.default.fontSize = `${fontSize}px`;
+
+      const fontFamily = this._associatedObject
+        .getProperties(this.project)
+        .get('fontFamily')
+        .getValue();
+      this._pixiObject.textStyles.default.fontFamily = fontFamily;
+
+      const wordWrap = this._associatedObject
+        .getProperties(this.project)
+        .get('wordWrap')
+        .getValue();
+      if (wordWrap !== this._pixiObject._style.wordWrap) {
+        this._pixiObject._style.wordWrap = wordWrap === 'true';
+        this._pixiObject.dirty = true;
+      }
+
+      const align = this._associatedObject
+        .getProperties(this.project)
+        .get('align')
+        .getValue();
+      if (align !== this._pixiObject._style.align) {
+        this._pixiObject._style.align = align;
+        this._pixiObject.dirty = true;
+      }
+
+      this._pixiObject.position.x =
+        this._instance.getX() + this._pixiObject.width / 2;
+      this._pixiObject.position.y =
+        this._instance.getY() + this._pixiObject.height / 2;
+      this._pixiObject.rotation = RenderedInstance.toRad(
+        this._instance.getAngle()
+      );
+
+      if (this._instance.hasCustomSize() && this._pixiObject) {
+        const customSize = this._instance.getCustomWidth();
```

Comment:
Prefer specific names: customSize => customWidth

Reply:
Rule of thumb: if a variable is the result of `getSomething`, then there is a good chance that you variable should be called `something`.